var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Warpx(MakefilePackage):
+    """WarpX is an advanced electromagnetic Particle-In-Cell code. It supports
+    many features including Perfectly-Matched Layers (PML) and mesh refinement.
+    In addition, WarpX is a highly-parallel and highly-optimized code and
+    features hybrid OpenMP/MPI parallelization, advanced vectorization
+    techniques and load balancing capabilities.
+    """
+
+    homepage = "https://ecp-warpx.github.io/index.html"
+    url      = "https://github.com/ECP-WarpX/WarpX"
+
+    version('master', git='https://github.com/ECP-WarpX/WarpX.git', tag='master')
+    version('dev', git='https://github.com/ECP-WarpX/WarpX.git', tag='dev')
+
+    depends_on('mpi', when='+mpi')
+
+    variant('dims',
+            default='3',
+            values=('1', '2', '3'),
+            multi=False,
+            description='Number of spatial dimensions')
+
+    variant('psatd', default=False, description='Enable PSATD solver')
+    variant('do_electrostatic', default=False, description='Include electrostatic solver')
+    variant('debug', default=False, description='Enable debugging features')
+    variant('tprof', default=False, description='Enable tiny profiling features')
+    variant('openmp', default=True, description='Enable OpenMP features')
+
+    resource(name='amrex',
+             git='https://github.com/AMReX-Codes/amrex.git',
+             tag='development',
+             destination='.')
+
+    resource(name='picsar',
```

Comment:
would be cool if future releases allow some kind of linking with an externally provided picsar (also since we have a package for it already in spack), but so far this is ok.